plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,237 @@
+ */
+import { entityRouteRef } from '../../routes';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { catalogApiRef } from '../../api';
+import React, { PropsWithChildren, useEffect } from 'react';
+import HoverPopover from 'material-ui-popup-state/HoverPopover';
+import {
+  bindHover,
+  bindPopover,
+  usePopupState,
+} from 'material-ui-popup-state/hooks';
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  Chip,
+  makeStyles,
+  Tooltip,
+  Typography,
+} from '@material-ui/core';
+import { Alert, Skeleton } from '@material-ui/lab';
+import EmailIcon from '@material-ui/icons/Email';
+import InfoIcon from '@material-ui/icons/Info';
+import { useApiHolder, useRouteRef } from '@backstage/core-plugin-api';
+import {
+  CompoundEntityRef,
+  isUserEntity,
+  isGroupEntity,
+  UserEntity,
+  GroupEntity,
+  Entity,
+} from '@backstage/catalog-model';
+import { Link, Progress } from '@backstage/core-components';
+
+/**
+ * Properties for an entity popover on hover of a component.
+ *
+ * @public
+ */
+export type EntityPeekAheadPopoverProps = PropsWithChildren<{
+  entityRef: CompoundEntityRef;
+}>;
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '30em',
+    },
+    descriptionTypography: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      display: '-webkit-box',
+      WebkitLineClamp: 2,
+      WebkitBoxOrient: 'vertical',
+    },
+  };
+});
+
+const maxTagChips = 4;
+
+const EmailCardAction = ({ email }: { email: string }) => {
+  return (
+    <Tooltip title={`Email ${email}`}>
+      <Button target="_blank" href={`mailto:${email}`} size="small">
+        <EmailIcon color="action" />
+      </Button>
+    </Tooltip>
+  );
+};
+
+const UserCardActions = ({ entity }: { entity: UserEntity }) => {
+  return (
+    <>
+      {entity.spec.profile?.email && (
+        <EmailCardAction email={entity.spec.profile.email} />
+      )}
+    </>
+  );
+};
+
+const GroupCardActions = ({ entity }: { entity: GroupEntity }) => {
+  return (
+    <>
+      {entity.spec.profile?.email && (
+        <EmailCardAction email={entity.spec.profile.email} />
+      )}
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+const EntityCardActions = ({ entity }: { entity: Entity }) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+
+  return (
+    <>
+      <Tooltip title="Show details">
+        <Link
+          component="button"
+          to={entityRoute({
+            name: entity.metadata.name,
+            namespace: entity.metadata.namespace || 'default',
+            kind: entity.kind.toLocaleLowerCase('en-US'),
+          })}
+        >
+          <InfoIcon color="action" />
+        </Link>
+      </Tooltip>
+    </>
+  );
+};
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+export const EntityPeekAheadPopover = ({
+  entityRef,
+  children,
+}: EntityPeekAheadPopoverProps) => {
```

Comment:
Being a bit picky, but could we make this `props: EntityPeekAheadPopoverProps`? And then unpack `const { entityRef, children } = props`. This shape makes its way into the API report and it looks a lot better without the unpacking, and doesn't start to leak internals once you want to rename them or add defaults or whatever.

Reply:
done.